csvapi/parser.py

```diff
@@ -46,17 +39,8 @@ def from_csv(filepath, encoding='utf-8', sniff_limit=SNIFF_LIMIT, agate_types=No
         except ValueError:
             kwargs['delimiter'] = ';'
             return agate.Table.from_csv(filepath, **kwargs)
-    except agate.CastError:
-        try:
-            kwargs = {
-                'sniff_limit': sniff_limit,
-                'encoding': encoding,
-                'column_types': agate_tester()
-            }
-            app.logger.warning('Types from csv-detective provoke errors, use of agate type tester instead.')
-            return agate.Table.from_csv(filepath, **kwargs)
-        except Exception as e:
-            app.logger.error('error casting %s', e)
+    except Exception as e:
+        app.logger.error('error casting %s', e)
```

Comment:
```suggestion
```

This except was supposed to be a fallback for a `CastError` and we don't handle this case anymore. I think we can let the generic exception bubble up as before.